### PR TITLE
Make refresh button execute correct query

### PR DESCRIPTION
**Pull request by enolal826**
I'm new to the project, so any suggestion would be appreciated. In case of complex discussions, I guess it would be better to chat on Slack.

This PR is a first approach to https://github.com/questdb/questdb/issues/1482. Looking at the code I found two approaches:
- `grid.js` having the responsibility of the last executed query (DISCARDED)
 The idea was to include a `lastQuery` parameter similar to the current `query` variable, which stores the last executed query.  However, this is discarded because the process of running the actual query happens in the editor.

- Editor storing the last executed query (IMPLEMENTED)
 In this approach, `grid.js` simply executes whatever query is passed and the responsibility is only on the editor side.
  
   Both running a new query and refreshing the current one depend on the `running` state, so I have included a new flag `isRefresh`.   It is also possible to keep this parameter in a different state action (even within the component, avoiding using Redux), but we would have to set the `running` and `isRefresh` parameters separately… I wonder if it would create any race condition in the future. Something like: 
```
 setIsRefresh(true); 
toggleRunning(); 
```
In order to retrieve the last executed query, my first idea was to use the notifications log, but we are storing React components, so it would more difficult to get the query string. Therefore, I have created a new variable `lastExecutedQuery`.

### CODE REVIEW

**Comment by insmac** [comment]
Hello, @enolal826 and thanks for your contribution! 👋 We do appreciate a thoroughly explained thought process on the issue fix you have implemented 🥇 

Regarding implementation #1 (`grid.js`), technically I think it could be done, since this file triggers a `grid.refresh` event that passes query in payload and the listener for it is defined within the editor. The problem is that the only thing this listener does is `toggleRunning`, so you would need to refactor this part either way. Because we're already in the process of refactoring grid and moving it to React (https://github.com/questdb/questdb/issues/1487), I don't think it's worth doing - your solution seems better.

I have tested your PR locally and it works as described beside two issues:

1. Query is executed twice. If I check Network tab, I'm seeing two consecutive `/exec` queries one by one for any given statement - to test, try creating a table and you'll see a `table already exists` error in the Console as it ran twice. This happens for both DQL and DDL as well.

2. Consider a following video:

https://user-images.githubusercontent.com/1871646/140721910-46b068fc-51a3-4c96-8a12-4e424b1f0aad.mp4

In short, it should refresh a query that has resulted in the result set displayed on the grid. In our case it will re-run an incorrect one that results in an error. I think we should save the query only if it has been a success, because we can assume that whatever you're seeing in the grid is a result of the last _successful_ DQL query performed.

Happy to jump on Slack to discuss further if you have any concerns.

**Comment by enolal826** [comment]
Thanks @insmac for your feedback.

I have fixed the duplicate request and I've change the code to only save the last query when the result if DQL.

Hope this works, let me know otherwise

**Review by insmac — APPROVED** [review]
@enolal826, thanks for the updates. The code works correctly - both issues are fixed on my end.
